src/contracts/core/Slasher.sol

```diff
@@ -1,102 +1,368 @@
+     * @param epoch the epoch to get the pending slashing rate for
+     *
+     * @return the parts per hundred million that will be slashed for the
+     * given operator, strategy, and epoch, across all operator set assuming
+     * no more modifications to requested slashing rate for the operator.
+     */
+    function getTotalPendingSlashingRate(
+        address operator,
+        IStrategy strategy,
+        uint32 epoch
+    ) external view returns (uint32) {
+        uint64 totalSlashingRate = slashingRequests[operator][strategy][epoch].slashingRate;
+        if (totalSlashingRate >= SlashingAccountingUtils.BIPS_FACTOR_SQUARED) {
+            totalSlashingRate = SlashingAccountingUtils.BIPS_FACTOR_SQUARED;
+        }
+        return uint32(totalSlashingRate);
+    }
 
-    function isFrozen(address) external view returns (bool) {}
+    /**
+     * @notice gets whether withdrawals of the given strategy delegated to the given operator can be withdrawn and the scaling factor
+     * @param operator the operator the withdrawal is delegated to
+     * @param strategy the strategy the withdrawal is from
+     * @param epoch the last epoch the withdrawal was slashable until
+     * @return whether the withdrawal can be executed
+     * @return whether there was a slashing request for the given operator and strategy at the given epoch
+     */
+    function getWithdrawabilityAndScalingFactorAtEpoch(
+        address operator,
+        IStrategy strategy,
+        uint32 epoch
+    ) external view returns (bool, uint64) {
+        bool __canWithdraw = true;
+        uint64 scalingFactor = SlashingAccountingUtils.SHARE_CONVERSION_SCALE;
+        (uint32 lookupEpoch, bool found) = _getLookupEpoch(operator, strategy, epoch);
+        if (found) {
+            __canWithdraw = _canWithdraw(operator, strategy, lookupEpoch);
+            scalingFactor = slashingRequests[operator][strategy][lookupEpoch].scalingFactor;
+        }
+        return (__canWithdraw, scalingFactor);
+    }
 
-    function canSlash(address, address) external view returns (bool) {}
+    /**
+     * @notice gets whether withdrawals of the given strategy delegated to the given operator can be withdrawn
+     * @param operator the operator the withdrawal is delegated to
+     * @param strategy the strategy the withdrawal is from
+     * @param epoch the last epoch the withdrawal was slashable until
+     * @return whether the withdrawal can be executed
+     */
+    function canWithdraw(address operator, IStrategy strategy, uint32 epoch) public view returns (bool) {
+        bool __canWithdraw = true;
+        (uint32 lookupEpoch, bool found) = _getLookupEpoch(operator, strategy, epoch);
+        if (found) {
+            // if the slashing request is not executed, the operator cannot withdraw
+            __canWithdraw = _canWithdraw(operator, strategy, lookupEpoch);
+        }
+        return __canWithdraw;
+    }
```

Comment:
do you maybe just want to make this function call `getWithdrawabilityAndScalingFactorAtEpoch` and then only return the boolean value?
then we can also eliminate the abstraction of `_canWithdraw`...